distribution/kubernetes/discovery-service.yaml

```diff
@@ -0,0 +1,15 @@
+# Headless service discovery
+apiVersion: v1
+kind: Service
+metadata:
+  name: discovery
+  labels:
+    app: discovery
+spec:
+  clusterIP: None
```

Comment:
Would it make sense to add `publishNotReadyAddresses: true` and then add readynessProbes to the searcher-statefulset? This way discovery udp:7282 will resolve all endpoints all the time but you can limit searcher-service to only the functioning searcher nodes. To avoid bad requests.

Reply:
Yes, I wasn't aware of `publishNotReadyAddresses`, but that makes total sense. I'm going to add liveness and readiness problems in a follow-up PR. I need to implement them in our services first :)

Also, would you mind looking at the checklist I wrote in #1658? I may have missed a few things.

Reply:
Checklist looks pretty good to me, I did a pr for an initial helm chart we've been using. 

https://github.com/quickwit-oss/helm-charts/pull/1

Its kind of just a start, or what we've been using to test quickwit with. I'm not 100% on the direction of everything and made a few assumptions about the future. Hopefully thats of some help with that checklist